config/impl/src/main/java/io/apicurio/common/apps/config/impl/DynamicConfigSource.java

```diff
@@ -63,11 +69,18 @@ public Set<String> getPropertyNames() {
     @Override
     public String getValue(String propertyName) {
         String pname = normalizePropertyName(propertyName);
-        if (configIndex.isPresent() && configIndex.get().hasProperty(pname) && storage.isPresent()) {
-            DynamicConfigPropertyDto dto = storage.get().getConfigProperty(pname);
-            if (dto != null) {
-                return dto.getValue();
+        if (configIndex.isPresent() && configIndex.get().hasProperty(pname)) {
```

Comment:
I think we should check here the `isReady` method, returning null (and thus continuing with the sources chain) if it's not ready yet, wdyt?

Reply:
Good point, I can add a check. Though there could be a situation where the storage becomes not ready, but we can still get the property from the cache.

Reply:
Well, you're relying on the fact that you know an implementation detail of an application that is using this library. IMO, the library itself should be agnostic to that detail.